src/main/java/io/fabric8/maven/docker/access/KeyStoreUtil.java

```diff
@@ -41,28 +43,40 @@ public static KeyStore createDockerKeyStore(String certPath) throws IOException,
         return keyStore;
     }
 
-    public static PrivateKey loadPrivateKey(String keyPath) throws IOException, GeneralSecurityException {
-        PEMKeyPair keyPair = loadPEM(keyPath);
-        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(keyPair.getPrivateKeyInfo().getEncoded());
-        return KeyFactory.getInstance("RSA").generatePrivate(keySpec);
+    static PrivateKey loadPrivateKey(String keyPath) throws IOException, GeneralSecurityException {
+        try (Reader reader = new FileReader(keyPath);
+            PEMParser parser = new PEMParser(reader)) {
+            Object readObject = parser.readObject();
+            if (readObject instanceof PEMKeyPair) {
+                PEMKeyPair keyPair = (PEMKeyPair) readObject;
+                return generatePrivateKey(keyPair.getPrivateKeyInfo());
+            } else if (readObject instanceof PrivateKeyInfo) {
+                return generatePrivateKey((PrivateKeyInfo) readObject);
+            }
+        }
+        return null;
```

Comment:
Thanks for the comments. That would be definitely more correct to throw exception here. Will fix.